Fruitties/shared/src/commonMain/kotlin/com/example/fruitties/viewmodel/ViewModelFactory.kt

```diff
@@ -0,0 +1,36 @@
+package com.example.fruitties.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewmodel.CreationExtras
+import androidx.lifecycle.viewmodel.MutableCreationExtras
+import androidx.lifecycle.viewmodel.initializer
+import androidx.lifecycle.viewmodel.viewModelFactory
+import com.example.fruitties.di.AppContainer
+
+val APP_CONTAINER_KEY = CreationExtras.Key<AppContainer>()
+
+/**
+ * Helper function to prepare CreationExtras.
+ *
+ * USAGE:
+ *
+ * val mainViewModel: MainViewModel = ViewModelProvider.create(
+ *  owner = this as ViewModelStoreOwner,
+ *  factory = MainViewModel.Factory,
+ *  extras = MainViewModel.newCreationExtras(appContainer),
+ * )[MainViewModel::class]
+ */
+fun creationExtras(appContainer: AppContainer): CreationExtras =
+    MutableCreationExtras().apply {
+        set(APP_CONTAINER_KEY, appContainer)
+    }
+
+inline fun <reified T : ViewModel> vmFactory(
```

Comment:
Can we pick a descriptive name for this factory? We could even do something like fruittiesViewModelFactory() to make it clear that this has app-specific choices.